src/pybind/mgr/cephadm/inventory.py

```diff
@@ -697,9 +723,69 @@ def save_host(self, host: str) -> None:
             j['scheduled_daemon_actions'] = self.scheduled_daemon_actions[host]
         if host in self.metadata_up_to_date:
             j['metadata_up_to_date'] = self.metadata_up_to_date[host]
+        if host in self.devices:
+            self.save_host_devices(host)
 
         self.mgr.set_store(HOST_CACHE_PREFIX + host, json.dumps(j))
 
+    def save_host_devices(self, host: str) -> None:
+        if host not in self.devices or not self.devices[host]:
+            logger.debug(f'Host {host} has no devices to save')
+            return
+
+        devs: List[Dict[str, Any]] = []
+        for d in self.devices[host]:
+            devs.append(d.to_json())
+
+        def byte_len(s: str) -> int:
+            return len(s.encode('utf-8'))
+
+        dev_cache_counter: int = 0
+        cache_size: int = self.mgr.get_foreign_ceph_option('mon', 'mon_config_key_max_entry_size')
+        if cache_size is not None and cache_size != 0 and byte_len(json.dumps(devs)) > cache_size:
+            # no guarantee all device entries take up the same amount of space
+            # splitting it up so there's one more entry than we need should be fairly
+            # safe and save a lot of extra logic checking sizes
+            cache_entries_needed = math.ceil(byte_len(json.dumps(devs)) / cache_size) + 1
+            dev_sublist_size = math.ceil(len(devs) / cache_entries_needed)
+            dev_lists: List[List[Dict[str, Any]]] = [devs[i:i + dev_sublist_size]
+                                                     for i in range(0, len(devs), dev_sublist_size)]
```

Comment:
Is this code covered by the UT? I mean, to check that we are splitting correctly the content into several entries. I think we have UT that checks the number of entries but it does not validate the content of each entry.

Reply:
I was referring that if we have a list of 5 elements i.e: `<dev-1><dev-2>....<dev-5>` and we have to split them into 3 entries, we make sure we get 3 lines and the content of the lines should be: 

```
<dev-1><dev-2>
<dev-2><dev-3>
<dev-5>
```
This way we check explicitly that we are breaking the lines correctly and we get the correct content on each one of them.

Reply:
added another case to this test to have five devices, each with different content, and make sure they get split up exactly as we'd expect.